src/error.rs

```diff
@@ -18,4 +18,7 @@ pub enum TokenGenerationError {
     /// Serde library error
     #[error("SerdeError occurred: {0}")]
     SerdeError(#[from] serde_json::Error),
+    /// The authentication service responded with an error
+    #[error("The authentication service returned an error:\nType: {0}, Message: {1}")]
+    AuthenticationError(String, String),
```

Comment:
Something i also have not done always but am now certain that it is better practice (for the future):
if your error/enum has multiple fields of the same type and there is no naural ordering (like x,y coordinates) i think it is much safer to use a Struct-like variant with named fields instead. this makes accidently swapping the members less likely (which would result in a wiered error message)